### PR TITLE
Refactor comments and other features

### PR DESCRIPTION
**Pull request by Slendy**
- Added Level comments and the ability to yay or boo all comments
- You can view level comments on the website as well
- Added proper comment deletion
- Deleted comments on the website appear the same as they do in-game
- Added the config option to disable reviews, level comments, and user comments
- Fixed the visual hash bug on the reset password page
- Fixed author labels not showing up in-game in certain scenarios

What still needs to be improved/added:
- The ability to post level comments and user comments from the website
- The ability to see comment ratings and boo or yay the comment

Before label fix:
Before

After label fix:
After

### CODE REVIEW

**Comment by Slendy** [comment]
@jvyden I think I messed up the migration because the tests failing is related to MySQL itself. Could you take a look?

**Comment by Slendy** [comment]
I guess the EF migration didn't add the correct annotations to the migration file so it wasn't being recognized and wouldn't get applied. No idea why that happened

File: ProjectLighthouse/Migrations/20220205132152_CommentRefactor.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                table: "Comments");
+
+            migrationBuilder.DropIndex(
+                name: "IX_Comments_TargetUserId",
+                table: "Comments");
+
+            migrationBuilder.RenameColumn(
+                name: "TargetUserId",
+                table: "Comments",
+                newName: "Type");
```
Comment (jvyden) [inline ProjectLighthouse/Migrations/20220205132152_CommentRefactor.cs:27]: Does this migrate existing comments properly? This doesn't seem right...

File: ProjectLighthouse/Pages/SlotPage.cshtml [hunk 2]
```diff
@@ -54,6 +57,36 @@
...
+    @if (Model.Comments.Count == 0)
+    {
+        <p>There are no comments.</p>
+    }
+
+    @foreach (Comment comment in Model.Comments!)
+    {
+        DateTimeOffset timestamp = DateTimeOffset.FromUnixTimeSeconds(comment.Timestamp / 1000);
+        StringWriter messageWriter = new();
+        HttpUtility.HtmlDecode(comment.getComment(), messageWriter);
```
Comment (jvyden) [inline ProjectLighthouse/Pages/SlotPage.cshtml:71]: This doesnt allow XSS does it?

**Review by jvyden — CHANGES_REQUESTED** [review]
Nice work!
As always, a few concerns...

**Comment by jvyden** [comment]
> I guess the EF migration didn't add the correct annotations to the migration file so it wasn't being recognized and wouldn't get applied. No idea why that happened

Perhaps the designer.cs file with the annotations (which we end up removing) didnt get generated? Not sure on that one, either.

File: ProjectLighthouse/Pages/SlotPage.cshtml (hunk 2 above)
Comment (Slendy) [reply ProjectLighthouse/Pages/SlotPage.cshtml:71]: It shouldn't, it's straight from the user class
test

File: ProjectLighthouse/Migrations/20220205132152_CommentRefactor.cs (hunk 1 above)
Comment (Slendy) [reply ProjectLighthouse/Migrations/20220205132152_CommentRefactor.cs]: No it doesn't, good catch.

**Review by jvyden — APPROVED** [review]
Just tested it out on my local dev instance, works well!

Thanks for the contribution!

**Pull request merged**